### DISCUSSION TITLE
Is there a contributor license agreement?

### DISCUSSION THREAD

**Discussion by leoluk** [minio/minio#16653] [Q&A] [answered]
I'm a bit confused about contributing to Minio.

Minio is AGPLv3-licensed and offers commercial licensing to customers who cannot comply with AGPL (nothing wrong with that - it's a well-established business model).

However, if a non-employee submits a PR to Minio, I don't see a CLA that would assign copyright to Minio, Inc. External contributors would retain copyright and Minio, Inc would not have the rights to license it to their commercial customers.

How does this work? Are there simply no external contributions, or are contributors asked to sign a CLA at some point?

**Comment by NyCodeGHG** [comment]
is there any clarification about this anywhere?

**Reply by klauspost** [reply, marked as answer]
https://github.com/minio/minio/blob/master/.github/PULL_REQUEST_TEMPLATE.md#community-contribution-license

**Reply by NyCodeGHG** [reply]
@klauspost thanks!
